p2panda-discovery/src/mdns/mod.rs

```diff
@@ -10,18 +10,22 @@ use std::time::Duration;
 
 use anyhow::Result;
 use flume::Sender;
-use futures_lite::StreamExt;
+use futures_lite::{FutureExt, StreamExt};
 use hickory_proto::rr::Name;
 use iroh_base::base32;
 use iroh_net::NodeAddr;
+use netwatch::netmon::Monitor;
+use tokio::sync::mpsc::{self, Receiver};
 use tokio_util::task::AbortOnDropHandle;
+use tracing::{debug, warn};
 
 use crate::mdns::dns::{make_query, make_response, parse_message, MulticastDNSMessage};
-use crate::mdns::socket::{send, socket_v4};
+use crate::mdns::socket::{send, socket_v4, socket_v4_unbound};
 use crate::{BoxedStream, Discovery, DiscoveryEvent};
 
 const MDNS_PROVENANCE: &str = "mdns";
 const MDNS_QUERY_INTERVAL: Duration = Duration::from_millis(1000);
+const SOCKET_REBIND_INTERVAL: Duration = Duration::from_millis(5000);
```

Comment:
We might consider adjusting this interval. Five seconds seems pretty reasonable for an ambient discovery service.